### PR TITLE
[CLI] `list-keymaps`/`list-layouts`: Check keyboard passed in

### LINKED ISSUE #15203: [Bug] `qmk list-layouts` and `list-keymaps` don't work
### Describe the Bug

```console
$ qmk list-layouts
<class 'TypeError'>
☒ unsupported operand type(s) for /: 'PosixPath' and 'NoneType'
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/milc/milc.py", line 526, in __call__
    return self.__call__()
  File "/usr/lib/python3.9/site-packages/milc/milc.py", line 531, in __call__
    return self._subcommand(self)
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/decorators.py", line 27, in wrapper
    return func(*args, **kwargs)
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/cli/list/layouts.py", line 16, in list_layouts
    info_data = info_json(cli.config.list_layouts.keyboard)
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/info.py", line 39, in info_json
    root_rules_mk = parse_rules_mk_file(cur_dir / keyboard / 'rules.mk')
TypeError: unsupported operand type(s) for /: 'PosixPath' and 'NoneType'

~/workspace/qmk
$ qmk list-keymaps
<class 'TypeError'>
☒ unsupported operand type(s) for /: 'PosixPath' and 'NoneType'
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/milc/milc.py", line 526, in __call__
    return self.__call__()
  File "/usr/lib/python3.9/site-packages/milc/milc.py", line 531, in __call__
    return self._subcommand(self)
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/decorators.py", line 27, in wrapper
    return func(*args, **kwargs)
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/cli/list/keymaps.py", line 16, in list_keymaps
    for name in qmk.keymap.list_keymaps(cli.config.list_keymaps.keyboard):
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/keymap.py", line 331, in list_keymaps
    rules = rules_mk(keyboard)
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/keyboard.py", line 150, in rules_mk
    keyboard = Path(resolve_keyboard(keyboard))
  File "/home/hugo/workspace/qmk/qmk_firmware/lib/python/qmk/keyboard.py", line 113, in resolve_keyboard
    rules = parse_rules_mk_file(cur_dir / keyboard / 'rules.mk')
TypeError: unsupported operand type(s) for /: 'PosixPath' and 'NoneType'
```

### Additional Context?

This is a clean setup with no changes or anything unusual locally.

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

If the CLI cannot infer the keyboard name (these commands have `@automagic_keyboard`) or it is not passed with `-kb`, show an error message.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #15203

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/cli/list/keymaps.py [hunk 1]
```diff
@@ -13,5 +13,9 @@
...
     """List the keymaps for a specific keyboard
     """
-    for name in qmk.keymap.list_keymaps(cli.config.list_keymaps.keyboard):
-        print(name)
+    if cli.config.list_keymaps.keyboard:
+        for name in qmk.keymap.list_keymaps(cli.config.list_keymaps.keyboard):
+            print(name)
+    else:
+        cli.log.error('Could not determine keyboard!')
+        return False
```
Comment (zvecr) [inline lib/python/qmk/cli/list/keymaps.py:21]:
```suggestion
    if not cli.config.list_keymaps.keyboard:
        cli.log.error('Missing required arguments: --keyboard')
        cli.subcommands['list-keymaps'].print_help()
        return False

    for name in qmk.keymap.list_keymaps(cli.config.list_keymaps.keyboard):
        print(name)
```

File: lib/python/qmk/cli/list/layouts.py [hunk 2]
```diff
@@ -13,6 +13,10 @@
...
-    info_data = info_json(cli.config.list_layouts.keyboard)
-    for name in sorted(info_data.get('community_layouts', [])):
-        print(name)
+    if cli.config.list_layouts.keyboard:
+        info_data = info_json(cli.config.list_layouts.keyboard)
+        for name in sorted(info_data.get('community_layouts', [])):
+            print(name)
+    else:
+        cli.log.error('Could not determine keyboard!')
+        return False
```
Comment (zvecr) [inline lib/python/qmk/cli/list/layouts.py:22]:
```suggestion
    if not cli.config.list_layouts.keyboard:
        cli.log.error('Missing required arguments: --keyboard')
        cli.subcommands['list-layouts'].print_help()
        return False

    info_data = info_json(cli.config.list_layouts.keyboard)
    for name in sorted(info_data.get('community_layouts', [])):
        print(name)
```

**Review by zvecr — COMMENTED** [review]
Maybe aligned with what lint does?

File: lib/python/qmk/cli/list/keymaps.py (hunk 1 above)
Comment (fauxpark) [reply lib/python/qmk/cli/list/keymaps.py:21]: Was referencing `compile`, but yeah, this is probably better.

**Review by zvecr — APPROVED** [review]

**Pull request merged**